packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.controller.js

```diff
@@ -2,13 +2,20 @@ const CONTAINER_NAME = 'pci.projects.project.public-gateways';
 
 export default class PublicGatewaysController {
   /* @ngInject */
-  constructor($translate, CucCloudMessage) {
+  constructor($translate, CucCloudMessage, coreURLBuilder) {
     this.$translate = $translate;
     this.CucCloudMessage = CucCloudMessage;
+    this.coreURLBuilder = coreURLBuilder;
   }
 
   $onInit() {
+    this.trackPublicGateways('public-gateway');
```

Comment:
It's strange tracking in the $onInit(). Either it's page tracking in which case do it via routing. Or it's click tracking, in which case do it on the function that calls this controller.

Reply:
updated as suggested.